packages/ember-model/lib/model.js

```diff
@@ -88,7 +88,7 @@ Ember.Model = Ember.Object.extend(Ember.Evented, {
 
     if (!reference) {
       reference = this.constructor._getOrCreateReferenceForId(id);
-      reference.record = this;
+      set(reference, 'record', this);
       this._reference = reference;
```

Comment:
Just out of curiosity, why use this syntax over 
```
set('reference.record', this);
```
Or does that lead to the same issue?